test/unit/factorization/test_compute_t_factor.cpp

```diff
@@ -399,39 +379,39 @@ void testComputeTFactor(comm::CommunicatorGrid grid, const SizeType a_m, const S
   // H_res = I - V T V*
   //
   // is computed and compared to the one previously obtained by applying reflectors sequentially
-  const auto& t = t_output.read(t_idx).get();
-  MatrixLocal<T> h_result = computeHFromTFactor(k, t, v);
+  const auto& t = t_output_h.read(t_idx).get();
+  MatrixLocal<T> h_result = computeHFromTFactor(k, t, v_local, v_start);
 
   is_orthogonal(h_result);
 
   // Note:
   // The error threshold has been determined considering that ~2*n*nb arithmetic operations (n*nb
   // multiplications and n*nb addition) are needed to compute each of the element of the matrix `h_result`,
   // and that TypeUtilities<T>::error indicates maximum error for a multiplication + addition.
-  SCOPED_TRACE(::testing::Message() << "Comparison test m = " << a_m << " n=" << a_n << " mb=" << mb
-                                    << " nb=" << nb << " k=" << k << " v_start=" << v_start);
+  SCOPED_TRACE(::testing::Message() << "Comparison test m=" << m << " k=" << k << " mb=" << mb
+                                    << " nb=" << nb << " v_start=" << v_start);
   const auto error = h_result.size().rows() * k * TypeUtilities<T>::error;
   CHECK_MATRIX_NEAR(h_expected, h_result, 0, error);
 }
 
 TYPED_TEST(ComputeTFactorTestMC, CorrectnessLocal) {
-  for (const auto& [a_m, a_n, mb, nb, k, v_start] : configs) {
-    testComputeTFactor<TypeParam, Backend::MC, Device::CPU>(a_m, a_n, mb, nb, k, v_start);
+  for (const auto& [m, k, mb, nb, v_start] : configs) {
+    testComputeTFactor<TypeParam, Backend::MC, Device::CPU>(m, k, mb, nb, v_start);
   }
 }
 
 TYPED_TEST(ComputeTFactorTestMC, CorrectnessDistributed) {
-  for (auto comm_grid : this->commGrids()) {
-    for (const auto& [a_m, a_n, mb, nb, k, v_start] : configs) {
-      testComputeTFactor<TypeParam>(comm_grid, a_m, a_n, mb, nb, k, v_start);
+  for (auto comm_grid : {this->commGrids()[0]}) {
```

Comment:
I imagine it is a debug leftover...

```suggestion
  for (auto comm_grid : this->commGrids()) {
```